src/channel/queuedChannel.ts

```diff
@@ -46,11 +46,13 @@ export class QueuedChannel<T> implements OutputChannel<T> {
 
         this.enqueue(message);
 
-        this.pending = this.pending.then(
-            () => this.channel
-                .publish(message)
-                .then(this.dequeue.bind(this)),
-        );
+        this.pending = this.pending
+            .catch(() => this.logger.debug('Failed to publish message, skipping...'))
+            .then(
+                () => this.channel
+                    .publish(message)
+                    .then(this.dequeue.bind(this)),
+            );
```

Comment:
Won't this jumble up the queue?

- Message A fails on the call to `publish`
- Message B is published due to the catch on the previous `pending`
- `this.dequeue` is called from the B message but will dequeue A
- B is left on the queue
- on next reload B is sent again

I think the catch should be between `publish` and `dequeue`